src/WinRT.Runtime/ObjectReference.cs

```diff
@@ -139,11 +139,12 @@ protected IObjectReference(IntPtr thisPtr)
             Dispose();
         }
 
-        public ObjectReference<T> As<
 #if NET
-        [DynamicallyAccessedMembers(DynamicallyAccessedMemberTypes.PublicFields)]
+        [RequiresUnreferencedCode(AttributeMessages.GenericRequiresUnreferencedCodeMessage)]
+        [Obsolete(AttributeMessages.GenericDeprecatedMessage)]
+        [EditorBrowsable(EditorBrowsableState.Never)]
 #endif
-        T>() => As<T>(GuidGenerator.GetIID(typeof(T)));
+        public ObjectReference<T> As<T>() => As<T>(GuidGenerator.GetIID(typeof(T)));
```

Comment:
For this and below, I am wondering if there is any backcompat risk by dropping the dynamicallyaccessedmembers attribute,

Reply:
I assume the vftbl dynamically accessed attributed we put as a fallback before takes care of all scenarios here?

Reply:
Which one are you thinking about? My thinking here was also that this method would not be producing a trim warning anyway, so people upgrading would still be warned. Effectively speaking you'd definitely want to use new projections in general when using trimming to get all the new improvements and bug fixes.

Reply:
My comment was in reference to this: https://github.com/microsoft/CsWinRT/blob/0b4bc36f92961979545cbc8b3869e4ddc202d12b/src/WinRT.Runtime/TypeExtensions.cs#L163-L176